src/test/java/com/spotify/docker/client/auth/ecr/ContainerRegistryAuthSupplierTest.java

```diff
@@ -0,0 +1,222 @@
+import com.amazonaws.util.Base64;
+import com.google.api.client.util.Clock;
+import com.spotify.docker.client.exceptions.DockerException;
+import com.spotify.docker.client.messages.RegistryAuth;
+import com.spotify.docker.client.messages.RegistryConfigs;
+import java.io.IOException;
+import org.hamcrest.FeatureMatcher;
+import org.hamcrest.Matcher;
+import org.joda.time.DateTime;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.Matchers;
+
+public class ContainerRegistryAuthSupplierTest {
+
+  @Rule
+  public final ExpectedException exception = ExpectedException.none();
+
+  private final DateTime expiration = new DateTime(2017, 5, 23, 16, 25);
+  private final String tokenValue = new String(encode("test-username:test-password".getBytes()));
+
+  private AmazonECR ecrClient = mock(AmazonECR.class);
+
+  private final AuthorizationData authData1 = new AuthorizationData()
+      .withAuthorizationToken(tokenValue).withExpiresAt(expiration.toDate())
+      .withProxyEndpoint("http://proxy");
+
+  private final Clock clock = mock(Clock.class);
+  private final int minimumExpirationSecs = 30;
+  private final EcrCredentials credentials = spy(new EcrCredentials(ecrClient, authData1));
+  private final ContainerRegistryAuthSupplier supplier = spy(new ContainerRegistryAuthSupplier(
+      ecrClient, clock, minimumExpirationSecs, credentials));
+
+  private static Matcher<RegistryAuth> matchesAccessToken(final AuthorizationData accessToken) {
+    String decoded = new String(Base64.decode(accessToken.getAuthorizationToken()));
+    final String username = decoded.split(":")[0];
+    final String password = decoded.split(":")[1];
+
+    final Matcher<RegistryAuth> usernameMatcher = new FeatureMatcher<RegistryAuth, String>(
+        is(username), "username", "username") {
+      @Override
+      protected String featureValueOf(final RegistryAuth actual) {
+        return actual.username();
+      }
+    };
+
+    final Matcher<RegistryAuth> passwordMatcher = new FeatureMatcher<RegistryAuth, String>(
+        is(password), "password", "password") {
+      @Override
+      protected String featureValueOf(final RegistryAuth actual) {
+        return actual.password();
+      }
+    };
+
+    return allOf(usernameMatcher, passwordMatcher);
+  }
+
+  @Before
+  public void before() {
+    GetAuthorizationTokenResult res = new GetAuthorizationTokenResult()
+        .withAuthorizationData(authData1);
+    doReturn(res).when(ecrClient).getAuthorizationToken(
+        Matchers.any(GetAuthorizationTokenRequest.class));
+  }
+
+  @Test
+  public void testAuthForImage_NoRefresh() throws Exception {
+    when(clock.currentTimeMillis()).thenReturn(
+        expiration.minusSeconds(minimumExpirationSecs + 1).getMillis());
+
+    assertThat(
+        supplier.authFor("1234567890.dkr.ecr.eu-west-1.amazonaws.com/foobar/barfoo:latest"),
```

Comment:
All of these image names seem to be the wrong format, to me. Shouldn't it be `1234567890.dkr.ecr.eu-west-1.amazonaws.com/foobar:latest`?